docbrown/core/src/tgraph.rs

```diff
@@ -301,54 +294,169 @@ impl TemporalGraph {
         )
     }
 
-    pub(crate) fn vertices(&self) -> Box<dyn Iterator<Item = VertexView<'_, Self>> + Send + '_> {
+    pub(crate) fn vertices(&self) -> Box<dyn Iterator<Item = VertexView> + Send + '_> {
         Box::new(
             self.adj_lists
                 .iter()
                 .enumerate()
                 .map(|(pid, v)| VertexView {
                     g_id: *v.logical(),
                     pid,
-                    g: self,
-                    w: None,
                 }),
         )
     }
 
     pub(crate) fn vertices_window(
         &self,
-        r: Range<i64>,
-    ) -> Box<dyn Iterator<Item = VertexView<'_, Self>> + Send + '_> {
+        w: Range<i64>,
+    ) -> Box<dyn Iterator<Item = VertexView> + Send + '_> {
         let unique_vids = self
             .index
-            .range(r.clone())
+            .range(w.clone())
             .map(|(_, vs)| vs.iter())
             .kmerge()
             .dedup();
 
         let vs = unique_vids.map(move |pid| match self.adj_lists[pid] {
-            Adj::Solo(lid) => VertexView {
-                g_id: lid,
-                pid,
-                g: self,
-                w: Some(r.clone()),
-            },
-            Adj::List { logical, .. } => VertexView {
-                g_id: logical,
-                pid,
-                g: self,
-                w: Some(r.clone()),
-            },
+            Adj::Solo(lid) => VertexView { g_id: lid, pid },
+            Adj::List { logical, .. } => VertexView { g_id: logical, pid },
         });
 
         Box::new(vs)
     }
 
-    pub(crate) fn neighbours(
+    pub(crate) fn edge(&self, v1: u64, v2: u64) -> Option<EdgeView> {
```

Comment:
As above, I think what you have created here is almost the undirected version. If the user asks for 1->2 we shouldn't return 2->1